tools/cli/helpers/doc-parser/src/class-doc-parser.php

```diff
@@ -104,15 +146,304 @@ function ( $item ) use ( $ignore ) {
+		foreach ( $phpdocnode->getTags() as $tag ) {
+			'@phan-var \PHPStan\PhpDocParser\Ast\PhpDoc\PhpDocTagNode $tag';
+
+			if ( $tag->name === '@param' && $tag->value instanceof InvalidTagValueNode ) {
+				$tag_value = $tag->value;
+				'@phan-var InvalidTagvalueNode $tag_value';
+
+				$pieces      = explode( ' ', $tag_value->value );
+				$type        = new IdentifierTypeNode( $pieces[0] );
+				$name        = $pieces[1] ?? 'argument';
+				$description = implode( ' ', array_slice( $pieces, 2 ) );
+				$tags[]      = new ParamTagValueNode( $type, true, $name, $description, false );
+			}
+		}
+
+		return $tags;
+	}
+
+	/**
+	 * Returns a list of PHP files in a folder, recursing into subfolders. Heavily inspired by
+	 * the WordPress PHPDoc parser.
+	 *
+	 * @see https://github.com/WordPress/phpdoc-parser/blob/7fc2227d2d4fb73f9f0b6e233413f3f9f9840e80/lib/runner.php#L17
+	 *
+	 * @param string $directory the folder to look in.
+	 *
+	 * @return array an array of filenames.
+	 * @throws \Exception $e If unable to traverse the filesystem.
+	 */
+	public function get_wp_files( $directory ) {
+		$iterableFiles = new \RecursiveIteratorIterator(
+			new \RecursiveDirectoryIterator( $directory )
+		);
+		$files         = array();
+
+		try {
+			foreach ( $iterableFiles as $file ) {
+				if ( 'php' !== $file->getExtension() ) {
+					continue;
+				}
+
+				$files[] = $file->getPathname();
+			}
+		} catch ( \UnexpectedValueException $exc ) {
+			throw new \Exception( sprintf( 'Directory [%s] contained a directory we can not recurse into', $directory ) );
+		}
+
+		return $files;
+	}
+
+	/**
+	 * Returns an array of docblock annotations for apply_filter function calls, with keys being names of filters
+	 * used.
+	 *
+	 * @param array $nodes  Parser node objects for hook calls.
+	 * @return array docblock annotations.
+	 */
+	public function get_filter_calls( $nodes ): array {
+
+		$blocks = array();
+
+		foreach ( $nodes as $node ) {
+
+			$arguments = $node->getArgs();
+			$hook_name = array_shift( $arguments );
+
+			// Purging any comments that could have been attributed to this argument.
+			$comments = $hook_name->getComments();
+			if ( ! empty( $comments ) ) {
+				$hook_name->setAttribute( 'comments', null );
+			}
+
+			$new_block = array(
+				'type'      => $node->name->name === 'apply_filters' ? 'filter' : 'action',
+				'line'      => $node->getLine(),
+				'end_line'  => $node->getEndLine() > 0 ? $node->getEndLine() : $node->getLine(),
+				'name'      => $this->pretty_print_hook_name( $hook_name ),
+				'arguments' => array(),
+			);
+
+			foreach ( $arguments as $argument ) {
+				$new_block['arguments'][] = $this->printer->prettyPrint( array( $argument ) );
+			}
+			$blocks[] = $new_block;
+		}
+
+		return $blocks;
+	}
+
+	/**
+	 * Pretty prints the name for the hook, taking an argument object as input.
+	 *
+	 * @suppress PhanRedundantConditionInLoop Phan is having trouble getting some of nikic/php-parser type definitions.
```

Comment:
This suppress seems mostly unnecessary. It's complaining below on line 438 because it knows the value already is an Expr, so it's wanting you skip checking for that. If you really want the check anyway, I'd probably suppress it inline down there with a comment about why you don't trust the typing. 😉